### DISCUSSION TITLE
How to create a post which contains a link to another Markdown file

### DISCUSSION THREAD

**Discussion by dphummel** [community/community#137984] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have successfully created my blog using GitHub Page.  In a post, I am able to successfully create a link to an external resource using 
“[text] (https://www.foobar.com/foo)”.   I am now trying to create a post (x.md) which links to a Markdown file in my local repository (y.md).  

I have placed the file x.md in the _posts directory and named it “2024-07-18-x.md”.  I have placed the y.md file in _posts directory naming it y.md.  I did not add the date to the file because I don’t want the file to appear as a post in my blog.  Within “2024-07-18-x.md”, I have added “[text] (x.md)”.  I have used x.md with the intent of Jekyll accessing the file from the current working directory (i.e. _posts).  When the html is generated, Jekyll is generating a link to an html, x.html, in the directory 2024/07 which does not exist because I did not name the x.md file “2024-07-18-y.md”.

When I name the file “2024-07-18-y.md” in _posts, I am able to link to the file using “[text] (y.html)” in “2024-07-18-x.md”.  However, this causes the y.md to appear as a post which I don’t want to have happen.

How can i create a valid link to a local Markdown file using GitHub Pages without having the Markdown file appearing as a post on my blog?

Thank you.

Note:  I have place a space between “]” and “(“ in the Markdown because I learned that this post is interpreting this Markdown.

**Comment by maryna-gol** [comment]
To link to a local Markdown file in your GitHub Pages blog without making it appear as a post, you need to move the file out of the `_posts` directory and place it somewhere else, like a dedicated folder for resources. Here's a step-by-step guide:

### Steps:

1. **Create a Separate Folder for Resources**:
   - Instead of placing `y.md` in the `_posts` directory, create a new directory in your repository for storing non-post files (e.g., `assets` or `resources`).

   For example:
   ```
   /assets/y.md
   ```

2. **Update the Link in Your Post**:
   - Now that the `y.md` file is in the `assets` folder, you can link to it from your post like this:
     ```markdown
     [text](assets/y.md)
     ```

3. **Accessing the Raw Markdown File**:
   - If you want to link to the raw Markdown file (not rendered as HTML), ensure you're linking to the raw file by using the correct relative path in the repository.

4. **Prevent the File from Appearing as a Post**:
   - By moving the file out of the `_posts` directory, you ensure that it won’t be treated as a blog post by Jekyll.

### Example:

- You create a post called `2024-07-18-x.md` in the `_posts` directory.
- Move the `y.md` file to `/assets/y.md`.
- In your post (`2024-07-18-x.md`), link to the `y.md` file like this:

   ```markdown
   [Click here to view the document](assets/y.md)
   ```

This way, the file won't be treated as a blog post, and you'll still be able to link to it from within your post. When the link is clicked, users will see the raw Markdown file or its rendered version depending on how GitHub Pages processes it.

**Reply by dphummel** [reply]
Maryan, I really appreciate your response!

I moved y.md into a directory at the same level as _posts called documents.  I updated 2024-07-18-x.md to reference y.md using the following:  [text] (../documents/y.md).  I had to use "../" to go up a directory since documents is at the same level as _posts.  When GitHub generates the HTML using Jekyll, the reference to y.md is still resolving to https://my_site/2024/07/y.md which does not exist.  It is not clear to be why Pages is looking for y.md in /2024/07.

Note:  When I preview x.md in GitHub and I select the link to y.md in x.md, the link properly displays the file y.md in preview mode.

Any thoughts on why might be causing this issue?

Thank you!

**Reply by dphummel** [reply]
Part of me wonders if the issue has to deal with the configuration of the repository I copied to create by blog - https://github.com/chadbaldwin/simple-blog-bootstrap/generate

**Reply by dphummel** [reply]
I decided to run a test by forking the [jeykll-now repo](https://github.com/barryclark/jekyll-now).  I am able to successfully use the approach you suggested to link to the y.md file in the x.md post!

I noticed when I hover over of the link to y.md in x.md, the url is https://dphummel.github.io/documents/y.md.  However, when I hover over the link to y.md using the repo I originally forked, the URL to y.md is notified to include the year and month of the x.md post (i.e., 2024/07).  I need to figure out what in the configuration of the original repo which I forked is causing this to happen.

**Reply by dphummel** [reply]
It seems like a permalink is being used to adjust the directories in the output, but I can't find the use of a Front Matter in any of the files which uses a permalink to adjust the output.  The repository is: 

https://github.com/dphummel/dphummel.github.io2

**Reply by dphummel** [reply, marked as answer]
I moved the original repository back to https://github.com/dphummel/dphummel.github.io.

I modified _config.yml by adding the line "permalink: none".  This change has removed the year/month from appearing in the URL of the site now.  And, the link to y.md in x.md works now.  I am not yet sure why I needed to add this line to the configuration file.
